rfc/rfc006-distributed-registry.md

```diff
@@ -0,0 +1,146 @@
+* **Organization**: a care organization exchanging data with other care organizations over a Nuts network.
+* **Vendor**: an entity developing software for care organizations participating in a Nuts network. Organizations
+  access the Nuts network through their vendor's software.
+* **Endpoint**: typically a URI or URL exposed by a vendor which can be used by other vendors to pull data.
+* **Service**: a grouping of endpoints implementing a service required by a Bolt.
+
+Other terminology comes from the [RFC001 Nuts Start Architecture](rfc001-nuts-start-architecture.md#nuts-start-architecture).
+
+## 3. Updating
+To update an object (vendor, organization, endpoint or service) the vendor MUST create a new document with the same identifier
+(uniquely identifiying the object) containing the new properties. Processing systems MUST accept the latest document
+as the current state of the object. Ordering MUST be performed according to RFC004.
+
+## 4. Documents
+Documents MUST be represented as JSON document as the JWS payload defined by [RFC004](rfc004-distributed-document-format.md).
+The **cty** header parameter from [RFC004](rfc004-distributed-document-format.md)) MUST contain the type as defined per document below.
+
+### 4.1. Vendor
+This document registers a vendor on the registry. The vendor ID MUST be embedded in the vendor CA certificates (see certificate structure RFC).
+All certificates for a vendor MUST contain the same vendor ID.
+
+#### Fields
+The following fields MUST be present:
+* **certs**: MUST be an array containing base64 encoded vendor CA certificates. There MUST be at least one entry.
+* **proof**: MUST contain cryptographic a proof that vouches for the vendor being a real software company. See section 5
+  for supported proof types.
+
+#### Document type
+The document type for registering vendors MUST be `registry/vendor`.
+
+#### Signer
+This document MUST be signed by the vendor itself.   
+
+### 4.2. Organization
+This document registers a care organization as a client of registered vendor (section 4.1).
+
+#### Fields
+The following fields MUST be present:
+* **id**: MUST be the organization's ID.
+* **vid**: (vendor ID) MUST be the organization's software vendor's ID as string.
+* **name**: MUST be the organization's name.
+* **proof**: MUST contain cryptographic a proof that asserts the organization as being an actual care organization.
+  at the same it serves as commitment from the care organization that it used the vendor's (identified by **vid**) software.
+  See section 6 for supported proof types.
+
+TODO: What if 2 vendors have the same organization as client?
+TODO: Do we need start/end dates now we don't have organization certs anymore?
+TODO: Should 'name' by an array to allow aliases?
+
+#### Document type
+The document type for registering organizations MUST be `registry/organization`.
+
+#### Signer
+This document MUST be signed by the vendor identified by **vendorID**.
```

Comment:
How to find the correct certificate and how does this work with past documents

Reply:
Signing certificate is present in the document (RFC004) so it can be verified against the truststore filled with all vendors certificates. What do you mean by past documents?

Reply:
If you say "MUST be signed by the vendor", you probably mean: "MUST be signed by the certificate currently in the truststore where the SAN matches the vendorID field...

If a Vendor CA gets a rekey, the current CA doesn't sign the older documents. If a new node replays the entire history, it'll have to replay it against older CA's.